micromanager_gui/main_window.py

```diff
@@ -215,10 +219,9 @@ def update_max_min(self, event=None):
 
         self.tab_wdg.max_min_val_label.setText(min_max_txt)
 
-    def snap(self):
+    def _snap(self):
         self.stop_live()
-
-        # snap in a thread so we don't freeze UI when using process local mmc
+        # update in a thread so we don't freeze UI
```

Comment:
I think we should simplifying the create worker here to just be:

```python
create_worker(self._mmc.snap, _start_thread=True)
```
and let the callbacks handle updating the viewer. That will allow other sources listening to the imageSnapped signal to pick it up.